### PR TITLE
prevent group plan users from purchasing subscription plans

### PR DESCRIPTION
**Pull request by tindNan**

### CODE REVIEW

File: src/routes/subscriptions/v2/payments/index.ts [hunk 1]
```diff
@@ -56,24 +58,34 @@ const paymentsRouter: FastifyPluginAsync = async (fastify): Promise<void> => {
...
+      const subType = await fastify.db.query.subscriptionType.findFirst({
+        where: eq(subscriptionType.id, req.body.subscription_type_id),
+      });
+
+      if (!subType) {
         throw fastify.httpErrors.notFound(
-          "Could not initiate purchase subscription for non-existent user",
+          "Could not find a subscription type with the given ID",
         );
       }
```
Comment (UwaoEvan) [inline src/routes/subscriptions/v2/payments/index.ts:91]: Don't forget to return the checkout_id

Comment (tindNan) [reply src/routes/subscriptions/v2/payments/index.ts:91]: oh yeah

will return that in a separate PR coming shortly

File: src/routes/subscriptions/v2/payments/index.ts [hunk 2]
```diff
@@ -56,24 +58,34 @@ const paymentsRouter: FastifyPluginAsync = async (fastify): Promise<void> => {
...
-          "Could not find a subscription type with the given ID",
+          "Could not initiate purchase subscription for non-existent user",
         );
       }
 
-      const user = await fastify.db.query.human.findFirst({
-        // @ts-ignore
-        where: eq(human.id, req.user.sub),
+      const existingGroupPlan = await fastify.db.query.groupPlan.findFirst({
+        where: eq(groupPlan.clientId, user.clientId),
```
Comment (tindNan) [inline src/routes/subscriptions/v2/payments/index.ts:73]: I think there might be a bug here.

Comment (tindNan) [reply src/routes/subscriptions/v2/payments/index.ts:73]: should be comparing with `existingUser.clientId`

Comment (UwaoEvan) [reply src/routes/subscriptions/v2/payments/index.ts:73]: True.

**Comment by tindNan** [comment]
@UwaoEvan made some updates here

**Pull request merged**